javascript/ql/src/semmle/javascript/frameworks/SocketIO.qll

```diff
@@ -84,165 +114,192 @@ module SocketIO {
   }
 
   /**
-   * Gets a data flow node that may refer to the socket.io namespace created at `ns`.
+   * A reference to a namespace object.
    */
-  private DataFlow::SourceNode namespace(NamespaceObject ns, DataFlow::TypeTracker t) {
-    t.start() and
-    exists(ServerNode srv |
-      // namespace lookup on `srv`
-      result = srv.getAPropertyRead("sockets") and
-      ns = srv.getServer().getDefaultNamespace()
-      or
-      exists(DataFlow::MethodCallNode mcn, string path |
-        mcn = srv.getAMethodCall("of") and
-        mcn.getArgument(0).mayHaveStringValue(path) and
-        result = mcn and
-        ns = MkNamespace(srv.getServer(), path)
+  class NamespaceBase extends SocketIOObject {
+    NamespaceObject ns;
+
+    NamespaceBase() {
+      exists(ServerObject srv |
+        // namespace lookup on `srv`
+        this = srv.ref().getAPropertyRead("sockets") and
+        ns = srv.getDefaultNamespace()
+        or
+        exists(DataFlow::MethodCallNode mcn, string path |
+          mcn = srv.ref().getAMethodCall("of") and
+          mcn.getArgument(0).mayHaveStringValue(path) and
+          this = mcn and
+          ns = MkNamespace(srv, path)
+        )
+        or
+        // invocation of a method that `srv` forwards to its default namespace
+        this = srv.ref().getAMethodCall(namespaceChainableMethod()) and
+        ns = srv.getDefaultNamespace()
       )
+    }
+
+    /** Gets the namespace to which this object refers. */
+    override NamespaceObject getNamespace() { result = ns }
```

Comment:
Do you need to have docstrings for these `getNameSpace` overrides? If you do, then try to make them more consistent (`refers` vs `belongs` vs `of`)

```
    /** Gets namespace belonging to this object. */

    /** Gets the namespace to which this object refers. */
    /** Gets the namespace to which this socket belongs. */
    /** Gets the default namespace of this server. */
```